django_mongodb/features.py

```diff
@@ -279,6 +284,10 @@ class DatabaseFeatures(BaseDatabaseFeatures):
             "update.tests.SimpleTest.test_empty_update_with_inheritance",
             "update.tests.SimpleTest.test_foreign_key_update_with_id",
             "update.tests.SimpleTest.test_nonempty_update_with_inheritance",
+            "model_fields.test_jsonfield.TestQuerying.test_join_key_transform_annotation_expression",
+            "model_fields.test_jsonfield.TestQuerying.test_order_grouping_custom_decoder",
```

Comment:
I think these last three don't actually do joins but the logic in `SQLCompiler._get_ordering()` is naive to the fact that `__` is a key transform. You can defer this to a separate commit or PR, but I would put these skips in a separate section that describes the issue.

Reply:
Mmh you are right, so for this one I think you should create a new PR. The if that check joins is wonky (only checks for __) I think it should be another short task. I will move those test to expected failures.